examples/fabric/build_your_own_trainer/trainer.py

```diff
@@ -4,11 +4,11 @@
 from typing import Any, cast, Iterable, List, Literal, Optional, Tuple, Union
 
 import torch
-from lightning_utilities.core import apply_to_collection, is_overridden
 from tqdm import tqdm
 
 import lightning as L
-from lightning.fabric.fabric import _unwrap_objects, Accelerator, Fabric, Logger, Strategy
+from lightning.fabric.fabric import _unwrap_objects, Accelerator, apply_to_collection, Fabric, Logger, Strategy
```

Comment:
This is importing apply_to_collection from a module that imports it. This is not a good idea. Why not keep the original `lightning_utilities` import?

Reply:
My point is that `apply_to_collection` is not defined in Fabric. Only imported.

So if that module stops using it, the import will get removed and then this will be a breaking change.

We don't guarantee BC for our internal imports. Otherwise we could only add imports and never remove or change them

Reply:
I understand your point, and I would personally also write the imports differently. But this was done like this, see the existing imports like Accelerator from `lightning.fabric.fabric`, before this PR. 

I changed the import here because the previous import was literally not working (ImportError), because apply_to_collection is not importable from lightning_utilities.core. So that's why I changed it, so that the example can run. 

It came from this (wrong) suggestion, which is my fault: https://github.com/Lightning-AI/lightning/pull/16938#discussion_r1124831360

Reply:
Then if you intend to make these imports "public API", please add comments in their definition so that we don't accidentally break the code of those who copy-pasted this